src/Core/maps/src/Handlers/MapPin/MapPinHandler.Android.cs

```diff
@@ -1,27 +1,61 @@
-﻿using Android.Gms.Maps;
+﻿using System;
+using Android.Gms.Maps;
 using Android.Gms.Maps.Model;
 using Microsoft.Maui.Handlers;
 
 namespace Microsoft.Maui.Maps.Handlers
 {
 	public partial class MapPinHandler : ElementHandler<IMapPin, MarkerOptions>
 	{
+		// Keep track of the actual marker associated with this handler using a weak reference
+		// to avoid potential memory leaks (the Marker is owned by the Google Maps view)
+		WeakReference<Marker>? _markerWeakReference;
+
+		internal Marker? Marker
```

Comment:
Add a `DisconnectHandler` override to properly clean up the weak reference when the handler is disconnected.

Reply:
@jsuarezruiz , Added DisconnectHandler. Please let me know if you have any concerns.